source/common/http/http2/protocol_constraints.cc

```diff
@@ -106,14 +97,14 @@ Status ProtocolConstraints::checkInboundFrameLimits() {
   }
 
   if (inbound_priority_frames_ >
-      static_cast<uint64_t>(max_inbound_priority_frames_per_stream_) * (1 + inbound_streams_)) {
+      static_cast<uint64_t>(max_inbound_priority_frames_per_stream_) * (1 + opened_streams_)) {
     stats_.inbound_priority_frames_flood_.inc();
     return bufferFloodError("Too many PRIORITY frames");
   }
 
   if (inbound_window_update_frames_ >
-      1 + 2 * (inbound_streams_ +
-               max_inbound_window_update_frames_per_data_frame_sent_ * outbound_data_frames_)) {
+      (10 * (1 + opened_streams_) +
+       (2 * max_inbound_window_update_frames_per_data_frame_sent_ * outbound_data_frames_))) {
```

Comment:
Could you change this to:
```
(2 * max_inbound_window_update_frames_per_data_frame_sent_ * (outbound_data_frames_ + 1))) {
```

An issue I see in the current formula is that setting max_inbound_window_update_frames_per_data_frame_sent_ to uint32 max does not effectively disable this flood check. It would be good to have a mechanism to mitigate any future issue in this logic by effectively disabling the check.

Reply:
Ok, I think this will work too.

Reply:
I've massaged it a bit more to accommodate cases where DATA frames have not yet been sent.